dvc/stage/__init__.py

```diff
@@ -367,7 +367,11 @@ def changed(self):
     def remove_outs(self, ignore_remove=False, force=False):
         """Used mainly for `dvc remove --outs` and :func:`Stage.reproduce`."""
         for out in self.outs:
-            if (out.persist or out.checkpoint or out.live) and not force:
+            if (
+                (out.persist and not out.append_only)
```

Comment:
This change would make it so that instead of being unprotected, the output is removed entirely (meaning it would be treated as a non-persistent output)